### ISSUE TITLE
How to Compare two ciphertexts?

### ISSUE THREAD

**Issue by vrush0022** [ibarrond/Pyfhel#80]
Is there any way I can compare two cyphertext?
I would like to perform c2a<c2b.

**Comment by ibarrond** [comment]
Unfortunately, Homomorphic Encryption (HE) does not support comparison between ciphertexts.

You can easily translate a comparison `ca > cb` into the extraction of the sign bit (MSB) of `ca-cb`. However, individual bit extraction/manipulation cannot be done with HE.

This problem is treated extensively in the literature, since this functionality is required for non-linear activation functions in Neural Networks, and other comparisons in Machine Learning. The two standard approaches to implement it are:
1. Use linear activation functions instead ([square](http://proceedings.mlr.press/v48/gilad-bachrach16.pdf), [polynomial approximations of RELU](https://arxiv.org/abs/1711.05189)).
2. Switch to Garbled Circuits for this computation (e.g.: [Gazelle](https://eprint.iacr.org/2018/073.pdf)).
3. Rely on Secret Sharing for this computation (e.g.: [SecureNN](https://www.researchgate.net/publication/334590018_SecureNN_3-Party_Secure_Computation_for_Neural_Network_Training)).

In short, you cannot compare two ciphertexts with Pyfhel.
